### ISSUE TITLE
Multiple Mapping Information reports incorrect mapping from NCBI Gene ID to UniProt

### ISSUE THREAD

**Issue by cmungall** [geneontology/panther-enrichment#19]
I entered a list of 500 GeneIDs, to my surprise >100 were reported as being ambiguous

E.g.

![image](https://user-images.githubusercontent.com/50745/69987439-82b6d600-150d-11ea-8da6-2dde80143b52.png)

I picked one of these, GeneID:219541 (https://www.ncbi.nlm.nih.gov/gene/219541 MED19)

Panther reports this as mapping to both https://www.uniprot.org/uniprot/P35790 (CHKA) and https://www.uniprot.org/uniprot/A0JLT2 (MED19), only the second appears to be correct

**Comment by RLovering** [comment]
I have just checked GeneID:1994, ELAVL1 ELAV like RNA binding protein 1:
UniProt mapping tool maps this to https://www.uniprot.org/uniprot/Q15717 (ELAVL1)
P06732 is a completely unrelated protein, Creatine kinase M-type
Both UniProt IDs are reviewed 

This is a very serious issue which is going to lead to misinterpretation of high-throughput data. I suggest you close the tool while this is resolved, 

Thanks

Ruth

@thomaspd @huaiyumi

**Comment by mugitty** [comment]
A majority of PANTHER users prefer to use a wide variety of protein ids, gene names, symbols, etc. This allows for a quick summary of the input list without having to map ids.

Many of these ids can be mapped to multiple genes or vice-versa where multiple ids can be mapped to a single gene. This is precisely why PANTHER provides the user with the multiple mapping information page with the long PANTHER gene id. This allows the user to avoid any ambiguity by updating any entries in the input list with the long PANTHER gene id.

**Comment by cmungall** [comment]
I conferred with @huaiyumi and @dustine32. As @mugitty says, there is ambiguity, in this case when entering an NCBI GeneID it is ambiguous as to whether this should be a GI or GeneID. This happens even when the user enters prefixed IDs, the prefix is discarded.

I proposed the following rules:

 - IF an ID prefix is provided using some kind of identifiable prefix (regardless of separator), THEN use that to disambiguate (never throw away prefixes if supplied)
 - IF the user specifies only numbers then ask what the database is (geneID, proteinID, ...) and use that
 - NEVER match using numbers alone. It is better to give an error than return highly problematic results

Additionally, maybe for now the safest thing is just to not us GI numbers at all?

**Comment by RLovering** [comment]
I agree with Chris, it is very worrying the GOC is promoting the use of a tool that is likely to be giving misinterpretations of a list of genes by excluding data or mapping it incorrectly.

**Comment by huaiyumi** [comment]
This issue stems from the Reference Proteome idmapping, which is the source data PANTHER uses. The string "GeneID:" is part of the ID (not a prefix), but RP idmapping removed it from their ids. That caused all the downstream issues observed here. We will figure out a way to handle it. Meanwhile, we will report it to RP. Thanks for the discussion to help us identify the problem.
